programs/stake/src/stake_state.rs

```diff
@@ -633,15 +633,57 @@ pub fn delegate(
     }
 }
 
+fn deactivate_stake(
+    invoke_context: &InvokeContext,
+    stake: &mut Stake,
+    stake_flags: &mut StakeFlags,
+    epoch: Epoch,
+) -> Result<(), InstructionError> {
+    let stake_history = invoke_context.get_sysvar_cache().get_stake_history()?;
```

Comment:
Can this be moved inside the `if` cases so it only affects the flagged accounts? No need to load this big sysvar if not needed.

Reply:
done.